File: server-replica/nginx/test/test-rate.lua
-- Streaming rate matches the configured bytes/second.
local p = require("probes")
for _, c in ipairs({ { "/_rate/slow", 4096 }, { "/_rate/fast", 10240 } }) do
    local r, err = p.read_for(c[1], 4)
    if not r then
        ngx.say(string.format("%-13s ERROR %s", c[1], err))
    else
        ngx.say(string.format("%-13s configured=%5d measured=%6.0f B/s bytes=%d",
            c[1], c[2], r.bytes / r.elapsed, r.bytes))
    end
end

File: server-replica/nginx/test/test-rate.js
// Streaming rate matches the configured bytes/second.
import probes from 'probes.js';

let cases = [
    { path: '/_rate/slow', configured: 4096 },
    { path: '/_rate/fast', configured: 10240 },
];

async function run(r) {
    let lines = [];
    for (let i = 0; i < cases.length; i++) {
        let c = cases[i];
        try {
            let result = await probes.readAll(c.path);
            let measured = Math.round(result.bytes / result.elapsed);
            lines.push(c.path + ' configured=' + c.configured +
                ' measured= ' + measured + ' B/s bytes=' + result.bytes);
        } catch (e) {
            lines.push(c.path + ' ERROR ' + e.message);
        }
    }
    r.return(200, lines.join('\n') + '\n');
}

export default { run };
